lib/EncodedFile.ml

```diff
@@ -0,0 +1,39 @@
+open Integers
+open Labels
+open ErrorUtils
+
+type data_file = {
+  data_bytes : Word.t Monoid.t;
+  data_label_mapping : DataLabel.t SMap.t;
+  data_label_position : MemoryAddress.t DataLabel.map;
+}
+
+type instr_file = {
+  instr_bytes : Word.t Monoid.t;
+  instr_label_mapping : ProgramLabel.t SMap.t;
+  instr_label_position : ProgramAddress.t ProgramLabel.map;
+}
+
+let write_data_file filename dfile =
+  if Sys.file_exists filename then
+    let txt =
+      Format.sprintf "The data segment output file '%s' already exists."
```

Comment:
Pour des fichiers de sorties, le comportement devrait peut être de réécrire par dessus. Pour référence, GCC et Clang ne se plaigne pas si le fichier .o existe déjà ou le fichier .exe.

Le soucis, c'est dès que vont recompiler on doit supprimer le fichier précédent ou alors appeler une sorte `make clean`.

Reply:
Sinon j'écrase avec un warning dans le cas où il existe déjà ?

Reply:
Va pour un warning ! C'est un bon compromis

Reply:
Fixed in ecf8949a2acd1b3f3c7509bc4fcc6ff07f12e557.